joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstCreator.scala

```diff
@@ -36,23 +41,30 @@ class AstCreator(
     with AstForLambdaCreator
     with AstGenNodeBuilder[AstCreator] {
 
-  protected val logger: Logger                                       = LoggerFactory.getLogger(classOf[AstCreator])
+  protected val logger: Logger = LoggerFactory.getLogger(classOf[AstCreator])
+  protected val tempAliasToNameSpaceMappingFilePath: Option[Path] =
+    tmpDir.map(dir => Paths.get(dir.pathAsString, s"alias-cache${UUID.randomUUID().toString}"))
   protected val methodAstParentStack: Stack[NewNode]                 = new Stack()
   protected val scope: Scope[String, (NewNode, String), NewNode]     = new Scope()
-  protected val aliasToNameSpaceMapping: mutable.Map[String, String] = mutable.Map.empty
-  protected val lineNumberMapping: Map[Int, String]                  = positionLookupTables(parserResult.fileContent)
-  protected val declaredPackageName = parserResult.json(ParserKeys.Name)(ParserKeys.Name).str
-  protected val fullyQualifiedPackage =
-    goMod.getNameSpace(parserResult.fullPath, declaredPackageName)
+  protected var aliasToNameSpaceMapping: mutable.Map[String, String] = mutable.Map.empty
```

Comment:
Line 49-56 is going to become technical debt and hard to maintain. Can we try to find a solution that avoids so much global mutable state in this class?

Reply:
Reading the code below, perhaps you want to look at what happens in `AstSummaryVisitor` in `RubySrc2Cpg` and `CSharpSrc2Cpg` for how to do a pre-parse without all this mutable state?

Reply:
At the very least, we need to move these variables to another trait, as it's becoming very difficult to follow this file.